pkg/identityserver/gateway_registry.go

```diff
@@ -728,6 +736,46 @@ func (is *IdentityServer) purgeGateway(ctx context.Context, ids *ttnpb.GatewayId
 	return ttnpb.Empty, nil
 }
 
+func (is *IdentityServer) batchDeleteGateways(
+	ctx context.Context,
+	req *ttnpb.BatchDeleteGatewaysRequest,
+) (*emptypb.Empty, error) {
+	if err := is.assertGatewayRights(
+		ctx,
+		req.GatewayIds,
+		&ttnpb.Rights{
+			Rights: []ttnpb.Right{ttnpb.Right_RIGHT_GATEWAY_DELETE},
+		},
+	); err != nil {
+		return nil, err
+	}
+	var (
+		err     error
+		deleted = make([]*ttnpb.GatewayIdentifiers, 0)
+	)
+	err = is.store.Transact(ctx, func(ctx context.Context, st store.Store) error {
+		deleted, err = st.BatchDeleteGateways(ctx, req.GatewayIds)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
```suggestion
		return err
```

Reply:
We don't do this with any other method that uses a transaction so I'll keep this as it is.